.github/action/setup-node-env/action.yml

```diff
@@ -0,0 +1,53 @@
+# this is a reusable workflow that will be called by the
+# push_staging and push_production workflows
+# more info: https://docs.github.com/en/actions/learn-github-actions/reusing-workflows
+
+name: gutenberg/setup-node-env
+description: Setup the Node.js environment including `pnpm` and dependencies
+
+inputs:
+    install:
+        default: true
+        description: Whether to install dependencies.
+    install-working-directory:
+        required: false
+        description: Optional working directory to run installation in.
+    frozen-lockfile:
+        default: false
```

Comment:
Why do we want to default it to false?

Reply:
I'm not sure! There were places that were doing just `npm install` rather than `npm ci` and I didn't want to change their behavior in this PR.

Reply:
In https://github.com/WordPress/gutenberg/pull/38624/commits/bd1d499c05c0a01cae5b3c157ca549130f551102 I changed it to just do a regular frozen-lockfile install everywhere relying on the default behavior in CI you mentioned. I can't think of a reason why CI should need to use a non-frozen install anywhere anyway.